wallet-new/src/Cardano/Wallet/API/V1/ReifyWalletError.hs

```diff
@@ -267,15 +268,13 @@ newTransactionError e = case e of
                 V1.TooBigTransaction
 
           ex@(CoinSelHardErrUtxoExhausted balance _payment) ->
-              case (readMaybe $ T.unpack balance) of
+              -- NOTE balance & payment are "prettified" coins representation (e.g. "42 coin(s)")
+              case (readMaybe $ T.unpack $ T.dropWhileEnd (not . C.isDigit) balance) of
```

Comment:
That's what I understood from this indeed. Any arguments about having `Coin` here instead of `Int` since it conveys more meaning. I don't see any reasons why we would do an early conversion when throwing the error.

Reply:
`Coin` is not applicable, because that is specific to the Cardano domain. Other domains will not use `Coin` for their representation of values.

Reply:
Ho true, forgot that the errors were generics here :+1: